src/codegen/compile.ml

```diff
@@ -7253,12 +7286,12 @@ module Serialization = struct
                We update the memo location here so that loops work
             *)
             get_thing ^^ set_result ^^
-            get_memo ^^ get_result ^^ store_word32 ^^
-            get_memo ^^ compile_add_const 4L ^^ Blob.lit env (typ_hash t) ^^ store_word32
+            get_memo ^^ get_result ^^ write_compressed_pointer env ^^
+            get_memo ^^ compile_add_const 4L ^^ Blob.lit env (typ_hash t) ^^ write_compressed_pointer env
```

Comment:
Oh wow, at first I didn't understand this code - I thought the type hash was just a u32, not a blob.

Reply:
Yes, it is complicated. The hash is a string (of potentially arbitrary length if I am not mistaken) and is thus captured in a blob. As a result, we need to compress that blob pointer too.